effectful/ops/syntax.py

```diff
@@ -547,19 +549,46 @@ def __hash__(self):
         return hash(self._default)
 
     def __default_rule__(self, *args: Q.args, **kwargs: Q.kwargs) -> "Expr[V]":
-        try:
+        from effectful.ops.semantics import await_
+
+        if self._is_async:
+            # For async operations, create an async wrapper
+            @functools.wraps(self)
+            async def async_default_rule():
+                try:
+                    try:
+                        # Call the async function and await it
+                        result: Coroutine = self._default(*args, **kwargs)  # type: ignore
+                        # Type ignore because we know it's async but the type system doesn't
+                        return await result
+                    except NotImplementedError:
+                        warnings.warn(
+                            "Operations should raise effectful.ops.types.NotHandled instead of NotImplementedError.",
+                            DeprecationWarning,
+                        )
+                        raise NotHandled
+                except NotHandled:
+                    # Return the term representation
+                    return typing.cast(
+                        Callable[Concatenate[Operation[Q, V], Q], Expr[V]], defdata
+                    )(await_, self, *args, **kwargs)
```

Comment:
It looks like this is the only place where `await_` is introduced, but I'm not sure if that's right? That may need to happen in `Operation.__call__` instead. We should clarify the invariants we're trying to maintain around `await_` and async `Operation`s.